crypto/property/defn_cache.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/err.h>
+#include <internal/propertyerr.h>
+#include <internal/property.h>
+#include <openssl/lhash.h>
+#include "property_lcl.h"
+
+/*
+ * Implement a property definition cache.
+ * These functions assume that they are called under a write lock.
+ * No attempt is made to clean out the cache, except when it is shut down.
+ */
+
+typedef struct {
+    const char *prop;
+    PROPERTY_LIST *defn;
+    char body[1];
```

Comment:
"unwarranted chumminess with the compiler" (web-search it)
Pay the price of portable code and make this a pointer.

Reply:
Double allocating and ruining cache coherency in code that's meant to be fast to avoid something that's always worked...

Reply:
I think that's a misguided/premature optimization, at the expense of more clear code. We're going to be dispatching to *crypto code* :)

Reply:
C99 allows using char body[]; Is C99 viable target for OpenSSL 3.0.0?

Reply:
We can't use C99.  However, this code is portable and safe.  There are two restrictions: don't use an array of such structures (no space) and don't store a type other than char there (alignment).